### ISSUE TITLE
[no-unnecessary-condition][bug] False positive match when chaining default value

### ISSUE THREAD

**Issue by bogdan** [typescript-eslint/typescript-eslint#2343]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": "error",
  }
}
```

```TS
const a = undefined;
const b = a || 'none';
```

**Expected Result**

No errors

**Actual Result**

```
/Users/bogdan/makabu/unstoppable/unstoppable-domains-website/default/zz.ts
  2:31  error  Unnecessary conditional, value is always truthy  @typescript-eslint/no-unnecessary-condition

✖ 1 problem (1 error, 0 warnings)
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.28.0` |
| `@typescript-eslint/parser`        | `2.28.0` |

**Comment by bradzacher** [comment]
are you using `strictNullChecks`?

**Comment by bogdan** [comment]
> are you using strictNullChecks?

Yes, I am.

That seem to be fixed in 3.x series. I think I am fine if we close it.

**Comment by bradzacher** [comment]
Please ensure you try the latest version before raising an issue.
